src/ipld/cid_vec.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::utils::cid::{CidVariant, BLAKE2B256_SIZE};
+use cid::{
+    multihash::{self, Code::Blake2b256},
+    Cid,
+};
+use fvm_ipld_encoding::DAG_CBOR;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+// CidVec takes advantage of the fact that the V1 DAG-CBOR Blake2b-256 variant (which can be stored in 32 bytes vs 96 bytes for a `Cid` type) is +99.99% of all CIDs. CidVec defaults to the `Vec<[u8; BLAKE2B256_SIZE]>` type, only using the more expensive `Vec<Cid>` type when necessary.
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum CidVec {
+    V1Cids(Vec<[u8; BLAKE2B256_SIZE]>),
```

Comment:
They're not just V1. They're V1+DAG_CBOR+Blake2b.

Reply:
While `V1DAGCBORBlake2b256Cids` is more accurate, it seems a bit clunky. Do you suggest I go with that naming convention anyway, or do you have suggestions for a better name?